security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -927,13 +944,8 @@ func (sc *SecretCache) shouldRotate(secret *security.SecretItem) bool {
 	return rotate
 }
 
-func (sc *SecretCache) isTokenExpired(secret *security.SecretItem) bool {
-	// skip check if the token passed from envoy is always valid (ex, normal k8s sa JWT).
-	if sc.configOptions.AlwaysValidTokenFlag {
-		return false
-	}
-
-	expired, err := util.IsJwtExpired(secret.Token, time.Now())
+func isTokenExpired(token string) bool {
+	expired, err := util.IsJwtExpired(token, time.Now())
```

Comment:
I think you are breaking authentication with '1st party' K8S token - which AFAIK still needs to be supported.

Any reason for removing the 'always valid token' feature ?

Reply:
Actually IsJwtExpired (https://github.com/istio/istio/blob/master/security/pkg/util/jwtutil.go#L35) returns false if "exp" claim does not exist. So we are ok for 1st party K8S Token case. There are test cases in jwtutil_test.go for 1st party k8s jwt.

Reply:
I don't think that point is reached - the K8S secrets are not JWT ( at least on my cluster ), so the first line of that method will return true ( unless I'm missing something ). 

But I think making that method return false - if it's not JWT, it's not expired - would work, and may be the simplest solution. Technically JWT without expiration are also not valid if I remember the RFC correctly, so the method name 
is correct - returns true if the parameter is a jwt that is expired, and false if it's not a jwt or not expired.